### ISSUE TITLE
7z should not expand for rar archive

### ISSUE THREAD

**Issue by bitstreamout** [scop/bash-completion#207]
There is a problem with the RAR support within 7z as this code is not legal ... compare with https://www.7-zip.org/license.txt and DOC/unRarLicense.txt from source 7z archive:

   3. The unRAR utility may be freely distributed. No person or company 
      may charge a fee for the distribution of unRAR without written
      permission from the copyright holder.

... that means all distribution have to remove the unRAR part of 7z

**Comment by andreabravetti** [comment]
But 7z can support unrar in some distro and I can't see the problem if it expand for rar and than 7z don't support it. On the other end it will be much more annoying if 7z support it and you stop expanding for rar.

**Comment by scop** [comment]
I agree. It would be best to ask 7z itself what formats it supports. I actually started that a long time ago but while the info is there, it's in a format that parsing from it seems really fragile and gave up on it (see `7z l`).
